public/app/services/settingsService.js

```diff
@@ -40,6 +45,200 @@ angular.module('kibibitCodeEditor')
+          'message': 'lineWrap should be boolean',
+          'currentValue': lineWrap,
+          'newValue': newValue
+        });
+        if (newValue !== lineWrap) {
+          if (settings.currentEditor) {
+            var session = settings.currentEditor.getSession();
+            session.setUseWrapMode(newValue);
+          }
+          lineWrap = newValue;
+        }
+      });
+
+      this.__defineGetter__('fontSize', function() {
+        return fontSize;
+      });
+
+      this.__defineSetter__('fontSize', function(newValue) {
+        console.assert(Number.isInteger(newValue), {
+          'message': 'fontSize should be integer',
+          'currentValue': fontSize,
+          'newValue': newValue
+        });
+
+        if (newValue !== fontSize) {
+          if (settings.currentEditor) {
+            var editor = settings.currentEditor;
+            editor.setFontSize(newValue);
+          }
+          fontSize = newValue;
+        }
+      });
+
+      this.__defineGetter__('tabWidth', function() {
+        return tabWidth;
+      });
+
+      this.__defineSetter__('tabWidth', function(newValue) {
+        console.assert(Number.isInteger(newValue), {
+          'message': 'tabWidth should be integer',
+          'currentValue': tabWidth,
+          'newValue': newValue
+        });
+
+        if (newValue !== tabWidth) {
+          if (settings.currentEditor) {
+            var session = settings.currentEditor.getSession();
+            session.setTabSize(newValue);
+          }
+          tabWidth = newValue;
+        }
+      });
+
+      this.__defineGetter__('isSoftTabs', function() {
+        return isSoftTabs;
+      });
+
+      this.__defineSetter__('isSoftTabs', function(newValue) {
+        console.assert(isBoolean(newValue), {
+          'message': 'isSoftTabs should be integer',
+          'currentValue': isSoftTabs,
+          'newValue': newValue
+        });
+
+        if (newValue !== isSoftTabs) {
+          if (settings.currentEditor) {
+            var session = settings.currentEditor.getSession();
+            session.setUseSoftTabs(newValue);
+          }
+          isSoftTabs = newValue;
+        }
+      });
+
+      this.__defineGetter__('isGutter', function() {
+        return isGutter;
+      });
+
+      this.__defineSetter__('isGutter', function(newValue) {
+        console.assert(isBoolean(newValue), {
+          'message': 'isGutter should be boolean',
+          'currentValue': isGutter,
+          'newValue': newValue
+        });
+
+        if (newValue !== isGutter) {
+          if (settings.currentEditor) {
+            var editor = settings.currentEditor;
+            editor.renderer.setShowGutter(newValue);
+          }
+        }
+        isGutter = newValue;
+      });
+
+      this.__defineGetter__('syntaxMode', function() {
+        return syntaxMode;
+      });
+
+      this.__defineSetter__('syntaxMode', function(newValue) {
+        console.assert(angular.isString(newValue), {
```

Comment:
I would suggest you'll check here that newValue is actually one of the modelist values.

`console.assert` is something that on production will be removed, and is here only to save us developers from making mistakes. Because of that, you shouldn't be afraid to check expensive things here.

So, a simple
```javascript
EDITOR.MODE_LIST.indexOf(newValue) !== -1 should do the trick
```